cocotb/regression.py

```diff
@@ -75,35 +79,18 @@ def _my_import(name):
 class RegressionManager:
     """Encapsulates all regression capability into a single place"""
 
-    def __init__(self, root_name, modules, tests=None, seed=None, hooks=[]):
+    def __init__(self, dut: SimHandle):
         """
         Args:
-            root_name (str): The name of the root handle.
-            modules (list): A list of Python module names to run.
-            tests (list, optional): A list of tests to run.
-                Defaults to ``None``, meaning all discovered tests will be run.
-            seed (int,  optional): The seed for the random number generator to use.
-                Defaults to ``None``.
-            hooks (list, optional): A list of hook modules to import.
-                Defaults to the empty list.
+            dut (SimHandle): The root handle to pass into test functions.
         """
-        self._queue = []
-        self._root_name = root_name
-        self._dut = None
-        self._modules = modules
-        self._functions = tests
+        self._dut = dut
         self._running_test = None
```

Comment:
Having both `_test` and `_running_test` members is confusing. Maybe:
```suggestion
        self._test_task = None
```

Reply:
Seems reasonable.